pagefaultgames/pokerogue
# [Bug] Fix final boss 2nd phase transition on status effect damage

## What are the changes?


Fixes #3219 

## What did change?
- Moves the final boss init method into a `BattleScene` method.
- `PostTurnStatusEffectPhase` now also checks for the init of phase 2 for final-boss  

## Screenshots/Videos



### Switching on burn damage

https://github.com/user-attachments/assets/09c09d2f-abc4-45d8-b465-0b6ce25f9bf4

### Still switching on direct damage too

https://github.com/user-attachments/assets/a643569b-3ddb-499c-a715-dd4b6a3b4e52

## How to test the changes?






```ts
const overrides = {
  STARTING_WAVE_OVERRIDE: 199,
  STARTING_LEVEL_OVERRIDE: 300,
  STARTER_SPECIES_OVERRIDE: Species.GYARADOS,
  MOVESET_OVERRIDE: [Moves.MOONBLAST, Moves.WILL_O_WISP],
  OPP_MOVESET_OVERRIDE: [Moves.SPLASH, Moves.SPLASH, Moves.SPLASH, Moves.SPLASH],
  STARTING_BIOME_OVERRIDE: Biome.END,
} satisfies Partial<InstanceType<typeof DefaultOverrides>>;
```
Start a new match and fight Eternatus.
Make sure to damage the last bit with BURN

## Checklist
    - ℹ️ After many hours of trying I wasn't able to write a test for the form change triggers. It seems that the way the Phases are injected is unsupported by the phase interceptor
